common/lib/util/eventemitter.ts

```diff
@@ -86,7 +88,7 @@ var EventEmitter = (function() {
 	 * @param listener (optional) the listener to remove. If not
 	 *        supplied, all listeners are removed.
 	 */
-	EventEmitter.prototype.off = function(event, listener) {
+	off(event: string | null, listener: Function) {
```

Comment:
I'm not sure this method signature matches the implementation, or there's something subtle going on that I don't understand. In the parlance of other languages, it looks like `off` is overloaded to allow just `listener` to be provided with no `event` name. I'm talking about the block starting with this statement, further down in the implementation of this method:

```js
if(typeof(event) == 'function') {
```

Reply:
Similar comment applies to previous, as Quintin correctly points out off is overloaded, so eg
```
off(listener?: Function): void;
off(event: null | string | string[], listener?: Function): void;
off(event?: null | string | string[] | Function, listener?: Function) {
```